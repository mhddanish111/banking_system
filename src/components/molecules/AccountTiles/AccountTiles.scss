@import "../../../styles/spacing.scss";
@import "../../../styles/colors.scss";
@import "../../../styles/boarder.scss";
@import "../../../styles/mixins.scss";
.account-tiles {
  flex-direction: column;
  border: $border-dark;
  border-radius: 10px;
  padding: $space-xxs;
  min-height: 200px;
  margin-left: $space-s;
  transition: 300ms;
  background: -webkit-linear-gradient(135deg, #53527ab0, #005788);
  transform-style: preserve-3d;
  transform-origin: 50% 50%;
  box-shadow: 10px 0 54px rgba(0, 0, 0, 0.55);
  font-size: 18px;
  color: $color-white;
  @include mobile {
      margin-bottom: $space-s;
      margin-left: 0;
  }
  a {
      color: $color-white;
  }
  .account-number {
    transform: translateZ(40px);
    font-family: kredit-front;
    font-weight: bold;
    font-size: 1.8rem;
    letter-spacing: 10px;
    z-index: 2;
    margin-bottom: $space-xxs;
    @include mobile {
      font-size: 1.25rem;
    }
    // width: calc(100% - 3.5rem);
    // text-align: center;
  }

  .account-details {
    display: flex;
    flex-direction: column;
    align-items: center;
  }
  
  .account-bottom {
    display: flex;
    border-top: $border-dark;
    padding-top: $space-x-small;
    margin-top: $space-xl;
  }
  
  .center-box {
    border-left: $border-dark;
    border-right: $border-dark;
  }
  
  .account-category-section {
      display: flex;
      margin-bottom: $space-s;
  }
  
  .account-type-section {
      display: flex;
      flex-direction: column;
      text-align: start;
  }
}


