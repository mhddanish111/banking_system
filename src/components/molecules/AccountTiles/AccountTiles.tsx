import React, { useMemo } from 'react';
import { Link } from 'react-router-dom';
import { split } from 'lodash';
import { Account } from '../../../type/AccountType';
import './AccountTiles.scss';

const AccountTiles: React.FC<Account> = React.memo(({ accounts }: Account) => {
  const { categoryName, accountNumber, amount, accountType } = accounts || {};
  const getCategory = (): string => {
    const categoryData = split(categoryName, ' ', 1);
    return categoryData[0];
  };
  return (
    accounts && (
      <div className="col-sm-3 col-md-4 col-lg-3 col-xs-12 account-tiles">
        <div className="col-xs-12 account-category-section">
          <div className="col-sm-6 col-xs-7 start-xs">
            <div className="box account-type-section">
              <span>{accountType}</span>
            </div>
          </div>
          <div className="col-sm-6 col-xs-5 end-xs">
            <span className="box">{getCategory()}</span>
          </div>
        </div>

        <div className="col-xs-12 middle-xs">
          <div className="box account-details">
            <span className="account-number">{accountNumber}</span>
            <span>Available Balance</span>
            <span>{`â‚¹ ${amount}`}</span>
          </div>
        </div>

        <div className="col-xs-12 account-bottom">
          <div className="col-sm-4 col-xs-5 start-xs">
            <Link to="/account" className="box">
              Statement
            </Link>
          </div>
          <div className="col-sm-4 col-xs-4 center-xs center-box">
            <Link to="/" className="box">
              Debit
            </Link>
          </div>
          <div className="col-sm-4 col-xs-3 end-xs">
            <Link to="/account" className="box">
              Pay
            </Link>
          </div>
        </div>
      </div>
    )
  );
});

export default AccountTiles;
