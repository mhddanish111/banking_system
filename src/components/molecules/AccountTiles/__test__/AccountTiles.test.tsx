import { shallow } from 'enzyme';
import AccountTiles from '../AccountTiles';

describe('<AccountTiles />', () => {
  const accounts = {
    type: 'Accounts',
    accountType: 'Savings Account',
    categoryName: 'Regular Account',
    accountNumber: 123456789,
    amount: 10000,
    debitCardCSVNumber: 123,
    debitCardNumber: 12312313123,
    expiryMonth: 1,
    expiryYear: 2023,
    limit: 0,
    maxAmount: 0,
    minAmount: 0,
    _id: '12345',
    charge: 0,
    users: [
      {
        _id: '12345',
        firstName: 'test',
        lastName: 'test',
      },
    ],
  };
  const wrapper = shallow(<AccountTiles accounts={accounts} />);
  it('should render the AccountTiles Component correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
