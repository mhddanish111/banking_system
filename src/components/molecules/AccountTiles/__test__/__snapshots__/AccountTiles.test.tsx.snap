// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AccountTiles /> should render the AccountTiles Component correctly 1`] = `
<div
  className="col-sm-3 col-md-4 col-lg-3 col-xs-12 account-tiles"
>
  <div
    className="col-xs-12 account-category-section"
  >
    <div
      className="col-sm-6 col-xs-7 start-xs"
    >
      <div
        className="box account-type-section"
      >
        <span>
          Savings Account
        </span>
      </div>
    </div>
    <div
      className="col-sm-6 col-xs-5 end-xs"
    >
      <span
        className="box"
      >
        Regular
      </span>
    </div>
  </div>
  <div
    className="col-xs-12 middle-xs"
  >
    <div
      className="box account-details"
    >
      <span
        className="account-number"
      >
        123456789
      </span>
      <span>
        Available Balance
      </span>
      <span>
        â‚¹ 10000
      </span>
    </div>
  </div>
  <div
    className="col-xs-12 account-bottom"
  >
    <div
      className="col-sm-4 col-xs-5 start-xs"
    >
      <Link
        className="box"
        to="/account"
      >
        Statement
      </Link>
    </div>
    <div
      className="col-sm-4 col-xs-4 center-xs center-box"
    >
      <Link
        className="box"
        to="/"
      >
        Debit
      </Link>
    </div>
    <div
      className="col-sm-4 col-xs-3 end-xs"
    >
      <Link
        className="box"
        to="/account"
      >
        Pay
      </Link>
    </div>
  </div>
</div>
`;
